cofmupy/config_parser.py

```diff
@@ -57,52 +63,53 @@ class ConfigParser:
         file_path (Union[str, Dict]): Path to the configuration file. Can also be
             a dictionary if the user prefers to provide the configuration directly
             without using a JSON file.
-        edge_sep (str): Edge separator for connections. Defaults to " -> ".
-        cosim_method (str): Method used to solve algebrauc loops. Defaults to "jacobi".
-        config_dict (Dict): The parsed configuration dictionary.
+        _config_object (ConfigObject): The parsed configuration object (type
+            ConfigObject)
         graph_config (Dict): Configuration for the graph engine.
         master_config (Dict): Configuration for the master solver.
         data_storages (Dict): Storage settings for external data.
         stream_handlers (Dict): Handlers for external data streams.
         error_in_config (bool): Indicates whether errors exist in the configuration.
     """
 
-    def __init__(
-        self,
-        file_path: Union[str, Dict],
-        edge_sep: str = " -> ",
-        cosim_method: str = "jacobi",
-        iterative: bool = False,
-    ) -> None:
+    def __init__(self, file_path: Union[str, Dict]) -> None:
         # Arguments
         self.file_path = file_path
 
-        self.config_dict: Dict = {}
+        self._config_object: ConfigObject
         self.graph_config: Dict = {}
         self.master_config: Dict = {}
         self.data_storages: Dict = {}
         self.stream_handlers: Dict = {}
         self.error_in_config: bool = False
 
-        # ------------ 1. Load config: self.config_dict ------------------
-        self.config_dict = self._load_config(file_check=True)
+        # ------------ 1. Load config: self._config_object ------------------
+        self._config_object = self._load_config(file_check=True)
 
         # ------------ 2. Apply defaults and perform validation ------------
-        self._apply_defaults(edge_sep, cosim_method, iterative)
         self._validate_configuration()
 
-        # ------------ 3. Prepend 'root' dir to present paths ------------
-        self._update_paths_in_dict()
-
-        # ------------ 4. Build configurations ---------------------------
+        # ------------ 3. Build configurations ---------------------------
+        self._build_storage_config()
         self._build_master_config()
         self._build_handlers_config()
         self._build_graph_config()
 
-    def _load_config(self, file_check: bool) -> Dict:
+    def get_config_dict(self):
```

Comment:
We could keep the `self.config_dict` as `@property` instead of adding this new function